crates/runtime-auth/src/api_key/mod.rs

```diff
@@ -46,6 +46,24 @@ impl HttpAuth for ApiKeyAuth {
     }
 }
 
+impl FlightBasicAuth for ApiKeyAuth {
+    fn validate(&self, _username: &str, password: &str) -> Result<String, Error> {
+        if self.api_keys.iter().any(|key| key == password) {
+            Ok(password.to_string())
+        } else {
+            Err(Error::InvalidCredentials)
+        }
+    }
+
```

Comment:
For learning purpose: Are password and bearer token in `ApiKeyAuth` both the same value as api_key? Why the 2 different notions (password, bearer token) are both used here?

Reply:
Yes, this is how our auth works in the Spice.ai Cloud Platform. It allows us to conform to the Flight Basic Auth specification, which can technically be different (i.e. correct user + password, get issued a token) - however for API key auth we don't want to have to issue and keep track of tokens.